src/cargo/ops/cargo_compile.rs

```diff
@@ -661,12 +661,34 @@ pub fn create_bcx<'a, 'cfg>(
                 continue;
             }
 
+            let guidance = if ws.is_ephemeral() {
+                if ws.ignore_lock() {
+                    "Try re-running cargo install with `--locked`".to_string()
```

Comment:
Just a comment for posterity, I'm a little uncomfortable using `is_ephemeral() && ignore_lock()` to detect that the user is running `cargo install`. That is true today, but who knows if that will change in the future? I'm going to approve with this since there really isn't any other way to detect this, and I don't feel it is too critical or likely to change any time soon. 

I suppose it might be worthwhile at some point to capture which command is currently being used for diagnostic purposes. I'd just be weary of anyone trying to use it for any other purpose which can cause problems.

Reply:
@ehuss I'd be happy to add a higher-level `is_cargo_install` function that currently delegates to these, by way of ensuring that if we need to change this we can do it in just one place.

Reply:
@ehuss https://github.com/rust-lang/cargo/pull/10966